GeoNames data augmentation

pvf2005 opened the pull request:
This is a data augmentation transformation making use of GeoNames. It will augment sentences containing cities or countries. More information in the README and examples are in test.json.

ashish3586 commented on transformations/geonames_transformation/transformation.py:
```diff
@@ -0,0 +1,272 @@
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
```
Please remove task "TEXT_TAGGING". This transformation is adding new words to a given sentence, so we can't use this for text tagging unless the generation method produces the respective tagging sequence along with the generated transformation.

pvf2005 replied:
You are right, the method is adding new words. "TEXT_TAGGING" is now removed.

kaustubhdhole commented:
I feel this is a great idea and can be powerfully used for data augmentation - it is clearly highly precise, and highly accurate from the looks of it, and the coverage is restricted by the names of all locations - and I think that itself covers quite a lot. There is a lot of information in GeoNames and you could make use of almost all of it. 
1. You have a pattern: "COUNTRY with capital" -> you could probably also add the reverse (if the city is a capital city) "city, the capital of COUNTRY"
2. countryInfo gives information about neighbours too - I guess you could add patterns like "COUNTRY, neighbouring COUNTRY2" or "COUNTRY, neighbouring COUNTRY2 & COUNTRY3", etc. or a better pattern something on those lines? I think you could use this to your advantage to amplify the amount of novelty since content on the internet would mostly follow the patterns that you've currently added. I think speaking about neighbouring countries is done and on rare occassions.
3. The country codes could also be used - Romania has a population... --> Romania (RO) has a population of.., Romania (RO/ROU) has a population of.., 
4. Also, to make it more accurate - while making the substitution, you could check if the words you are adding aren't already present in the sentence. eg. The UK, having capital London, is largely populated -->  The UK, with capital London, having capital London is largely populated. 
5. Also, please make a section in the README for Data and code provenance

Overall, nice work!

pvf2005 commented:
The suggestions from @kaustubhdhole were very interesting (especially with regard to the neighboring countries) and are now implemented. Additionally I added patterns for the spoken language(s) available in a country and the code is now easier to read (each pattern has its own method).

kaustubhdhole commented on transformations/geonames_transformation/README.md:
```diff
@@ -0,0 +1,45 @@
+
+
+## Data and code provenance
+This transformation makes use of city and country information obtained from GeoNames ( http://download.geonames.org/export/dump/ ).
```
Also mention the license of Geonames: https://github.com/google/BIG-bench/blob/83a7a0dc23f41b13daeffc84f5bc8346bffd7163/bigbench/benchmark_tasks/spelling_bee/README.md

pvf2005 replied:
The license is now added

kaustubhdhole commented:
Thank you very much for your changes. Your changes are merged. NLP has largely focussed on high-resource languages and domains. While this is not mandatory, if possible, we also encourage you to add a transformation in your mother tongue and/or a low-resource language which might need more attention from the NLP community. 

Thanks for your changes!
